dv_utils/datasets/contract_quality_check.py

```diff
@@ -0,0 +1,115 @@
+"""
+This module checks quality of a dataset in a data source for checks defined by the data contract.
+"""
+import logging
+import yaml
+
+import duckdb
+
+from ..settings import Settings
+from ..settings import settings as default_settings
+
+from ..log_utils import audit_log, LogLevel
+
+from ..client import Client
+
+from ..connectors.connector import populate_configuration
+from ..connectors import s3
+
+from .contract import Contract
+
+from soda.scan import Scan
+
+logger = logging.getLogger(__name__)
+
+
+class ContractQualityCheck:
+    NAMING_CONVENTION_MODEL="{{model}}"
+
+    def __init__(self):
+        self.client= Client()
+        self.contract=Contract()
+  
+    def check_contract_for_data_descriptor(self, collaboration_space_id: str, data_descriptor_id: str,data_connector_config_path:str):
+        logger.info(f"Get data descriptor for collaboration_space_id:{collaboration_space_id} - data_descriptor_id:{data_descriptor_id}")
+        data_descriptor=self.client.get_data_descriptor(collaboration_space_id,data_descriptor_id,self.client.DATA_PROVIDER_COLLABORATOR_ROLE_VALUE)
+        if data_descriptor != None:
+            self.contract.create_contract(data_descriptor,collaboration_space_id,data_descriptor_id,data_connector_config_path)
+            self.__check_contract(self.contract,data_connector_config_path)
+        else:
+            logger.error(f"No data descriptor available for collaboration_space_id:{collaboration_space_id} - data_descriptor_id:{data_descriptor_id}")
+            raise RuntimeError("Unable to check data contract")
```

Comment:
no exceptions in cage, or they should be catched whenever the method is used (and maybe should be specified then that it can raise an exception)

Reply:
✅